libs/editor/egui_editor/src/cursor.rs

```diff
@@ -186,24 +186,23 @@ impl Cursor {
     pub fn advance_word(
         &mut self, backwards: bool, buffer: &SubBuffer, segs: &UnicodeSegs, galleys: &Galleys,
     ) {
-        let (mut galley_idx, mut cursor) = galleys.galley_and_cursor_by_char_offset(self.pos, segs);
+        let mut galley_idx = galleys.galley_at_char(self.pos, segs);
         let galley = &galleys[galley_idx];
         let galley_text = galley.text(buffer);
         let galley_text_range = galley.text_range(segs);
 
         let word_bound_indices_with_words = galley_text.split_word_bound_indices();
         let word_bound_indices = word_bound_indices_with_words
             .clone()
-            .map(|(idx, _)| segs.byte_offset_to_char(DocByteOffset(idx)))
+            .map(|(idx, _)| {
+                segs.byte_offset_to_char(galley.byte_range().start + RelByteOffset(idx))
+            })
```

Comment:
bug was caused by confusion between offsets into galleys vs offsets into document, so I adjusted to work purely in terms of offsets into document